nibabel/streamlines/tests/test_tractogram.py

```diff
@@ -233,6 +248,20 @@ def test_getitem(self):
             assert_arrays_equal(sdict[-1][k], v[-1])
             assert_arrays_equal(sdict[[0, -1]][k], v[[0, -1]])
 
+    def test_extend(self):
+        total_nb_rows = DATA['tractogram'].streamlines.total_nb_rows
+        sdict = PerArraySequenceDict(total_nb_rows, DATA['data_per_point'])
```

Comment:
It is related to my previous comment. I meant I succeeded in reducing code duplication.